examples/bigcommerce/pages/api/cart.ts

```diff
@@ -0,0 +1,146 @@
+          body: JSON.stringify({
+            line_items: [JSON.parse(req.body).line_item],
+            channel_id: 1,
+          }),
+        },
+      )
+
+      if (!response.ok) throw new Error(response.statusText)
+
+      const result: RestResponse<CartResponse> = await response.json()
+
+      return res.status(200).json(result)
+    }
+
+    const response = await fetch(
+      `https://api.bigcommerce.com/stores/${config.bigcommerce.storeHash}/v3/carts/${req.query.cartId}/items`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'X-Auth-Token': config.bigcommerce.accessToken,
+        },
+        body: JSON.stringify({
+          line_items: [JSON.parse(req.body).line_item],
+          channel_id: 1,
+        }),
+      },
+    )
+
+    if (!response.ok) throw new Error(response.statusText)
+
+    const result: RestResponse<CartResponse> = await response.json()
+
+    return res.status(200).json(result)
+  }
+
+  if (req.method === 'PUT') {
+    if (req.query.cartId == null) {
+      return res
+        .status(400)
+        .json({ message: "Failed to update lineItem. 'cartId' query param required." })
+    }
+
+    if (req.query.lineItemId == null) {
+      return res
+        .status(400)
+        .json({ message: "Failed to update lineItem. 'lineItemId' query param required." })
+    }
+
+    const response = await fetch(
+      `https://api.bigcommerce.com/stores/${config.bigcommerce.storeHash}/v3/carts/${req.query.cartId}/items/${req.query.lineItemId}`,
+      {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json',
+          'X-Auth-Token': config.bigcommerce.accessToken,
+        },
+        body: req.body,
+      },
+    )
+
+    if (!response.ok) throw new Error(response.statusText)
+
+    const result: RestResponse<CartResponse> = await response.json()
+
+    return res.status(200).json(result)
+  }
+
+  if (req.method === 'DELETE') {
+    if (req.query.cartId == null) {
+      return res
+        .status(400)
+        .json({ message: "Failed to delete lineItem. 'cartId' query param required." })
+    }
+
+    if (req.query.lineItemId == null) {
+      return res
+        .status(400)
+        .json({ message: "Failed to delete lineItem. 'lineItemId' query param required." })
+    }
+
+    const response = await fetch(
+      `https://api.bigcommerce.com/stores/${config.bigcommerce.storeHash}/v3/carts/${req.query.cartId}/items/${req.query.lineItemId}`,
+      {
+        method: 'DELETE',
+        headers: {
+          'Content-Type': 'application/json',
+          'X-Auth-Token': config.bigcommerce.accessToken,
+        },
+      },
+    )
+
+    if (!response.ok) throw new Error(response.statusText)
+
+    if (response.status === 204) return res.status(204).json({})
+
+    const result: RestResponse<CartResponse | null> = await response.json()
+
+    return res.status(200).json(result)
+  }
+
+  return res.status(405)
```

Comment:
I think that if you don't send the response the request will just hang until the client times out. You might want to add `.send('Method Not Allowed')` or something to make sure the request doesn't hang

Reply:
You are correct. Thanks for catching that.